sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -943,3 +983,43 @@ final class RuleNotFoundExprMeta[INPUT <: Expression](
   override def convertToGpu(): GpuExpression =
     throw new IllegalStateException("Cannot be converted to GPU")
 }
+
+sealed abstract class Optimization
```

Comment:
This is already a pretty large file, so we may want to move optimizer-related rules/classes to a separate file, especially since it's expected to grow over time.